substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java

```diff
@@ -141,7 +171,15 @@ public Properties getProperties() {
 
     protected String getProperty(String key) {
         initializeLazyValue(key);
-        return properties.getProperty(key);
+        String ret = properties.getProperty(key);
+        if (ret == null && isNotSupported(key)) {
+            System.out.println("Java system property " + key + " is not available in native image. Please explicitly assign the expected value via -D" + key + "=, or avoid using this property.");
```

Comment:
The VM must never itself print any unsolicited output. So I don't see a way to print a warning message like this.

Reply:
Fixed by using exception here